toc.md

```diff
@@ -650,42 +818,32 @@ module Workerholic
 end
 ```
 
-Now we have what data we wanted to track and store. Next we needed to ask how we should store our data? We decided to use Redis because there's no need for a new dependency, very efficient writes and reads (in-memory store), automatic persistence to disk, and have the tools available to store serialized jobs, like using a sorted set.
+We add serialized jobs to the `workerholic:stats` sorted sets in redis with the `completed_at` timestamp as a score. This way when we want to retrieve jobs for the last 10 seconds we simply have to get all the jobs with a score between the current timestamp and the timestamp from 10 seconds ago.
 
 ##### How much data?
-once we had a foundation for live data, we should think about how much data we should store. Initially, we decided we do not need to store live data, and to just poll new data every 10 seconds. This posed a problem though once we introduced graphs into the web UI; just polling for new data and throwing away stale data was no longer an option. A quick-fix for this was to just store data on the front-end for a certain number of data points to create the graph. This worked, but only if the user stayed on the page. If the user navigated away from the page. The data would've been lost. Instead, we looked at Redis to store this data, up to 1000 seconds, for a total of 100 data points. Currently, our graphs only show up to 240 seconds, so this many data points is unnecessary, but it may become necessary if we decided to cover more time with our graphs.
 
-#### Historical Statistics
-Live data, check! Next, we want to display historical, and first we had to think about what type of data to store?
+Initially, we decided not to store live data and to just poll new data every 10 seconds. Once we introduced graphs into the web UI just polling for new data and throwing away stale data was no longer an option.
 
+At first, we decided to store data on the front-end for a certain number of data points so we could generate graphs. This worked, but only if the user stayed on the page. If the user navigated away from the page, the data be lost. Instead, we looked at Redis to store this data, up to 1,000 seconds, so we could have a total of 100 data points.
```

Comment:
the data was lost. Instead, we decided to rely on Redis to store this data, up to 1,000 seconds, so we could have a total of 100 data points.